app/src/main/java/com/jienihou/outfitxchange/HomeScreen.kt

```diff
@@ -15,11 +23,52 @@ import androidx.fragment.app.Fragment
  * create an instance of this fragment.
  */
 class HomeScreen : Fragment() {
+    private lateinit var bottomNavigationView: BottomNavigationViewEx
+    private lateinit var categoryAdapter: CategoryRVAdapter
+
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View? {
-        // Inflate the layout for this fragment
-        return inflater.inflate(R.layout.fragment_home_screen, container, false)
+        val view = inflater.inflate(R.layout.fragment_home_screen, container, false)
+
+        // Logo bar
+         val toolbar = view?.findViewById(R.id.toolbar) as Toolbar
+        (activity as AppCompatActivity).setSupportActionBar(toolbar)
+
+        // Categories Grid Recycler View
+        val categoryRV = view.findViewById(R.id.rv_categories) as RecyclerView
+        val columns = 2
+        categoryRV.layoutManager = GridLayoutManager(this.context, columns)
+        categoryAdapter = CategoryRVAdapter()
+        categoryRV.adapter = categoryAdapter
+        addCategories()
+
+        // Bottom Navigation View
+        bottomNavigationView = view.findViewById(R.id.bottom_navigation) as BottomNavigationViewEx
+        bottomNavigationView.enableItemShiftingMode(false)
+        bottomNavigationView.setTextVisibility(false)
+        bottomNavigationView.setOnNavigationItemSelectedListener { item ->
+            when (item.itemId) {
+                R.id.action_home -> Navigation.findNavController(view)
```

Comment:
Can we extract out `Navigation.findNavController(view)` seems to be shared amongst all the cases of the when statement.